src/PhpParser/Node/NodeFactory.php

```diff
@@ -337,19 +337,29 @@ public function createReturnBooleanAnd(array $newNodes): ?Expr
         return $this->createBooleanAndFromNodes($newNodes);
     }
 
-    public function createReprintedExpr(Expr $expr): Expr
+    /**
+     * Setting all child nodes to null is needed to avoid reprint of invalid tokens
+     * @see https://github.com/rectorphp/rector/issues/8712
+     *
+     * @template TNode as Node
+     *
+     * @param TNode $node
+     * @return TNode
+     */
+    public function createReprintedNode(Node $node): Node
     {
         // reset original node, to allow the printer to re-use the expr
-        $expr->setAttribute(AttributeKey::ORIGINAL_NODE, null);
+        $node->setAttribute(AttributeKey::ORIGINAL_NODE, null);
```

Comment:
comment doc can be updated to Node instead of Expr on line 351

```diff
-  // reset original node, to allow the printer to re-use the expr
+  // reset original node, to allow the printer to re-use the Node
```